pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -461,6 +461,11 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 
 // DeleteMachine deletes an AWS instance
 func (a *Actuator) DeleteMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	if machine.Labels["sigs.k8s.io/cluster-api-machine-type"] == "master" {
```

Comment:
Can you please use existing `IsMaster` func? and take `"sigs.k8s.io/cluster-api-machine-role"` into consideration as well for the func return value. See https://github.com/openshift/installer/blob/master/pkg/asset/machines/aws/machines.go#L54
Also we might want to consider to add `node-role.kubernetes.io/master: ""` labels to the backing master/workers machines so the machine would actually become source of truth for the node role https://github.com/openshift/machine-api-operator/blob/master/cmd/nodelink-controller/main.go#L391